docs/configuration/side-outputs.md

```diff
@@ -0,0 +1,97 @@
+# Side Outputs
+
+*alpha* - this may change in backwards-incompatible ways in the future. it has not be widely used in production scenarios yet.
+
+The **Side Outputs** feature allows you to configure additional outputs from your connector, to fulfill a couple of use cases:
+  - desired content is too large/long to be processed in a single synchronous request from client --> proxy --> source and back
+  - target API has rate limits; and you have another use for the desired response content
+  - having extra copy of all data coming through the proxy, pre- or post-processing
+
+If a **side output** is configured for an instance, a copy of each API response will be written to the side output. By default, this
+will be the copy of the content AFTER it has been processed by the proxy, but you can also configure it to be the original content
+exactly as returned by the source API.
+
+By default, the response body will also be returned to the client (caller) of the proxy. But if client (caller) sends a
+`X-Psoxy-No-Response-Body` header with request, then the proxy will not send a response body back to the client. Instead, it will
+return just the status code / headers back to the client.
+
+<!-- TODO : add data flow diagram of this case here -->
+
+
+## Configuration
+
+Use the following configuration properties to configure the side outputs from a proxy instance in API connector mode. These should
+be set as environment variables in the proxy instance:
+  - `SIDE_OUTPUT_ORIGINAL`/`SIDE_OUTPUT_SANITIZED` - defines the target of the side output. Eg `s3://bucket-name/`, `gs://bucket-name/`, etc;
+      - in the future, we might support `bq://dataset.table`, etc.
+      - `ORIGINAL`/`SANITIZED` - the **stage** of the data to be written to the side output.
+        - `ORIGINAL` - the original response from the source API, before any processing by the proxy
+        - `SANITIZED` - the response after it has been processed by the proxy (eg, sanitized, transformed, etc)
+
+
+Setting this up in terraform:
+
+Approaches:
+    1. `custom_side_outputs` in gcp/aws host modules; add key ==> configuration to that variable.
+
+Use case: you want a back-up of the data, or to have all the requests and responses otherwise available.
+
+```hcl
+custom_side_outputs = {
+    "zoom" = {
+        'ORIGINAL'  = "s3://my-bucket/original-outputs/"
+        'SANITIZED' = "s3://my-bucket/side-outputs/"
+    }
+}
+```
+   2. connector specs. a `provision_side_output` in the connector spec, which will cause the bucket to be provisioned.
+
+Use case: some responses from source API expected to be too large/too slow to return directly to the client, so you want to write them to a side output instead.
+
+```hcl
+provision_side_output = true
+```
+
+## Details
+For bucket use-cases, each response will be written as a single object in the bucket, with the following naming convention:
+
+`GET_api.google.com/v1/some/endpoint_{SHA256(query-params)}`
```

Comment:
this should also contain TIMESTAMP / Request time. Same calls on different days should be differentiable.
Even in the case of retries, probably we should have all attempts logged too?